File: 377/solution.py
class Solution(object):
    def combinationSum4(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        dp = []
        for i in range(target+1):
            dp.append(0)
        dp[0] = 1
        for i in range(target):
            for j in nums:
                if i + j <= target:
                    dp[i+j]+=dp[i]
        return dp[target]

File: 377/solution.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) {
        unsigned long long dp[target+1];
        for(int i=1; i<=target; ++i) dp[i] = 0;
        dp[0] = 1;
        for(int i=0; i<=target; ++i){
            for(auto j:nums){
                if(i-j>=0){
                    dp[i] += dp[i-j];
                }
            }
        }
        
        return dp[target];
    }
};
